File: web/lib/api/modules/dividend.ts
import { apiGet } from "@/lib/api/core/http";

export type MyDividendSummary = {
  total_rcb: string;
  month_rcb: string;
  latest_amount_rcb: string | null;
  latest_dividend_date: string | null;
  latest_status: string | null;
  current_svip_level: string | null;
  eligible: boolean;
};

export type MyDividendRecord = {
  id: number;
  dividend_date: string | null;
  svip_level_code: string;
  amount_rcb: string;
  amount_usdt: string;
  status: string;
  paid_at: string | null;
};

export type MyDividendRecordsResponse = {
  items: MyDividendRecord[];
  total: number;
  page: number;
  page_size: number;
};

export async function getMyDividendSummary(): Promise<MyDividendSummary> {
  return apiGet<MyDividendSummary>("/dividend/my/summary");
}

export async function getMyDividendRecords(
  page: number = 1,
  pageSize: number = 20,
): Promise<MyDividendRecordsResponse> {
  const params = new URLSearchParams({
    page: String(page),
    page_size: String(pageSize),
  });
  return apiGet<MyDividendRecordsResponse>(`/dividend/my/records?${params.toString()}`);
}

File: backend/app/schemas/dividend.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel

from app.schemas.response import ApiResponse


class MyDividendSummaryOut(BaseModel):
    total_rcb: str
    month_rcb: str
    latest_amount_rcb: Optional[str] = None
    latest_dividend_date: Optional[str] = None
    latest_status: Optional[str] = None
    current_svip_level: Optional[str] = None
    eligible: bool = False


class MyDividendRecordOut(BaseModel):
    id: int
    dividend_date: Optional[str] = None
    svip_level_code: str
    amount_rcb: str
    amount_usdt: str
    status: str
    paid_at: Optional[str] = None


class MyDividendRecordsOut(BaseModel):
    items: list[MyDividendRecordOut]
    total: int
    page: int
    page_size: int


class MyDividendSummaryApiResponse(ApiResponse[MyDividendSummaryOut]):
    pass


class MyDividendRecordsApiResponse(ApiResponse[MyDividendRecordsOut]):
    pass
